src/main/java/com/keniu/services/OrderServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package com.keniu.services;
+
+import com.keniu.dto.CreateOrderRequestDto;
+import com.keniu.dto.OrderDto;
+import com.keniu.dto.OrderItemDto;
+import com.keniu.dto.UpdateOrderRequestDto;
+import com.keniu.exceptions.EntityNotFoundException;
+import com.keniu.mappers.OrderItemMapper;
+import com.keniu.mappers.OrderMapper;
+import com.keniu.models.CartItem;
+import com.keniu.models.Order;
+import com.keniu.models.OrderItem;
+import com.keniu.models.ShoppingCart;
+import com.keniu.models.Status;
+import com.keniu.models.User;
+import com.keniu.repositories.OrderRepository;
+import com.keniu.repositories.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    @Transactional
```

Comment:
I remember our discussion about @Transactional in your prev PR, so I have investigated it and you can use it on class level and it will be enough